packages/frontend-2/lib/common/composables/toast.ts

```diff
@@ -1,60 +1,95 @@
-import { useTimeoutFn } from '@vueuse/core'
 import type { Optional } from '@speckle/shared'
 import type { ToastNotification } from '@speckle/ui-components'
+import { useTimeoutFn } from '@vueuse/core'
 import { ToastNotificationType } from '@speckle/ui-components'
 import { useSynchronizedCookie } from '~/lib/common/composables/reactiveCookie'
+import { v4 as uuidv4 } from 'uuid'
 
 /**
  * Persisting toast state between reqs and between CSR & SSR loads so that we can trigger
  * toasts anywhere and anytime
  */
 const useGlobalToastState = () =>
-  useSynchronizedCookie<Optional<ToastNotification>>('global-toast-state')
+  useSynchronizedCookie<Optional<ToastNotification[]>>('global-toast-state')
 
 /**
  * Set up a new global toast manager/renderer (don't use this in multiple components that live at the same time)
  */
 export function useGlobalToastManager() {
+  type Timeout = {
+    id: string
+    stop: () => void
+  }
+
   const stateNotification = useGlobalToastState()
 
-  const currentNotification = ref(stateNotification.value)
-  const readOnlyNotification = computed(() => currentNotification.value)
+  const timeouts = ref<Timeout[]>([])
+  const currentNotifications = ref<ToastNotification[]>(
+    Array.isArray(stateNotification.value) ? stateNotification.value : []
+  )
+  const readOnlyNotification = computed(() => currentNotifications.value)
 
-  const dismiss = () => {
-    currentNotification.value = undefined
-    stateNotification.value = undefined
+  // Remove a specific notification from the state
+  const removeNotification = (id: string) => {
+    const index = currentNotifications.value.findIndex((n) => n.id === id)
+    if (index !== -1) {
+      currentNotifications.value.splice(index, 1)
+      // Clean up timeout
+      timeouts.value = timeouts.value.filter((t) => t.id !== id)
+    }
   }
 
-  const { start, stop } = useTimeoutFn(() => {
-    dismiss()
-  }, 4000)
+  // Create a timeout for a notification
+  const createTimeout = (notification: ToastNotification) => {
+    const { stop } = useTimeoutFn(() => {
```

Comment:
https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?pvs=4#2a7cf25d251e403fac1ab53946887102

composable invoked in async scope (on mouse click, presumably)

Reply:
Yeah it's used on click and whenever the timeout is finished, do you have a suggestion regarding a better approach?

Reply:
Why not use real setTimeout (in CSR)?